miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/security/AuthorizedPaginatedDataSource.java

```diff
@@ -0,0 +1,45 @@
+package uk.ac.bbsrc.tgac.miso.service.security;
+
+import java.io.IOException;
+import java.util.List;
+
+import uk.ac.bbsrc.tgac.miso.core.security.SecurableByProfile;
+import uk.ac.bbsrc.tgac.miso.core.util.PaginatedDataSource;
+
+/**
+ * Filters a data source based on read access.
+ */
+public class AuthorizedPaginatedDataSource<T extends SecurableByProfile> implements PaginatedDataSource<T> {
+
+  private final PaginatedDataSource<T> backingSource;
+  private final AuthorizationManager authorizationManager;
+
+  public AuthorizedPaginatedDataSource(PaginatedDataSource<T> backingSource, AuthorizationManager authorizationManager) {
+    super();
+    this.backingSource = backingSource;
+    this.authorizationManager = authorizationManager;
+  }
+
+  @Override
+  public long count() throws IOException {
+    return backingSource.count();
+  }
+
+  @Override
+  public List<T> list(int offset, int limit, boolean sortDir, String sortCol) throws IOException {
+    return authorizationManager.filterUnreadable(backingSource.list(offset, limit, sortDir, sortCol));
```

Comment:
Nope. It's also not handled now on the list pages. Without putting authorization checks into Hibernate, I don't know who we'd do it.